tartiflette_aiohttp/_handler.py

```diff
@@ -36,21 +47,28 @@ def prepare_response(data):
 async def _handle_query(
     req, query, query_vars, operation_name, context_factory
 ):
-    context = await context_factory(req)
+    context_factory_mgr = context_factory(req)
 
-    try:
-        if not operation_name:
-            operation_name = None
-
-        return await req.app["ttftt_engine"].execute(
-            query=query,
-            variables=query_vars,
-            context=context,
-            operation_name=operation_name,
-        )
-    except Exception as e:  # pylint: disable=broad-except
-        logger.exception(e)
-        return {"data": None, "errors": _format_errors([e])}
+    # backwards compatibility with versions <= 1.2.0
+    # if context_factory is not a context manager, assume it is a coroutine function,
+    # and wrap it in a null context manager so it works with "async with" statement.
```

Comment:
I think we can get rid of the backward compat, the change is not that complex, with a good documentation and a 1.3.x version, it should be good enough.
OR
If it is possible to put this block at init time, so it's not called for each requests, then it would be more okay. If it's not possible then it's not bothering me to break this compatibility. IMHO not many people defined their own "context_factory".

Reply:
I've removed the backwards compatibility, so now context_factory is only accepted as an asynccontextmanager. I've also updated the type annotations accordingly.

Note that by defining the default_context_manager as an asynccontextmanager, there is now a runtime dependency on the async_generator package for Python 3.6.